### PR TITLE
Rm react test renderer (psammead)

### LINKED ISSUE #774: Remove use of react-test-renderer
We currently have 2 test renderers in Psammead and Simorgh - `react-test-renderer` and `@testing-library/react`.

Making use of 2 test renderers will lead to confusion about which one to use and when, especially when they have different ideas about how to write tests.

## Solution
Remove `react-test-renderer` and all use of it. Refactor shallow render tests because `@testing-library/react` does not have or encourage shallow rendering.

---

Some of the benefits of using `@testing-library/react` include:
 - encourages better testing practices and writing more maintainable tests 
 - is designed to enable and encourage writing tests that use components as the end users use them
 - it supports querying for elements in ways that encourage accessibility in components

Find out more about [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro).

This article briefly describes and compares React test renders https://codeburst.io/revisiting-react-testing-in-2019-ee72bb5346f4

Here is more on why [`@testing-library/react` doesn't do shallow rendering](https://kentcdodds.com/blog/why-i-never-use-shallow-rendering).

### PR DESCRIPTION
**Pull request by mykeels**
Resolves https://github.com/bbc/psammead/issues/774

Duplicate of https://github.com/bbc/psammead/pull/2405

**Overall change:** _Remove reference to react-test-renderers._

**Code changes:**

- _Remove react-test-renderer from package.json._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by thekp — CHANGES_REQUESTED** [review]
Your merge latest into this branch seems to have messed up your changelog changes so you will need to bump the package.json version/update changelog again :/

**Review by EinsteinNjoroge — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 | Version | Description |
 |---------|-------------|
 | 1.8.22 | [PR#2413](https://github.com/bbc/psammead/pull/2413) Talos - Bump Dependencies - @bbc/psammead-locales |
+| 1.8.21 | [PR#2405](https://github.com/bbc/psammead/pull/2405) Remove react-test-renderer |
```
Comment (rhenshaw56) [inline CHANGELOG.md]: :eyes:

**Review by rhenshaw56 — APPROVED** [review]
LGTM! the changelog needs a modification

**Comment by mykeels** [comment]
~The only change to the psammead-test-helpers package, was to the README file. Does this still require a version bump? @thekp @rhenshaw56~

**Review by thekp — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
   },
   "devDependencies": {
     "@babel/cli": "^7.6.4",
-    "@babel/core": "^7.6.4",
+    "@babel/core": "^7.6.2",
```
Comment (FK78) [inline package.json:46]: You've downgraded the dependency, it was bumped in this PR #2452.

File: package.json [hunk 3]
```diff
@@ -98,21 +98,20 @@
     "jest-fetch-mock": "^2.1.2",
     "jest-styled-components": "^6.3.3",
     "json5": "^2.1.1",
-    "lerna": "^3.18.1",
-    "lint-staged": "^9.4.2",
+    "lerna": "^3.17.0",
```
Comment (FK78) [inline package.json:101]: You've downgraded the dependency, it was bumped in this PR #2452.

**Review by FK78 — CHANGES_REQUESTED** [review]
Need to bump these dependencies before merging.

**Pull request closed without merging**